packages/rule-https-only/src/rule.ts

```diff
@@ -0,0 +1,179 @@
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class HttpsOnlyRule implements IRule {
+    private targetIsServedOverHTTPS: boolean = false;
+
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.security,
+            description: `sonarwhal rule that checks if your site is using HTTPS and if it has mixed content.`
+        },
+        id: 'https-only',
+        schema: [],
+        scope: RuleScope.site
+    }
+
+    public constructor(context: RuleContext) {
+        let target: string;
+        const reportedUrls: Set<string> = new Set();
+
+        const validateTarget = async (fetchEvent: FetchEnd): Promise<void> => {
+            const { resource } = fetchEvent;
+
+            if (!isHTTPS(resource)) {
+                debug('HTTPS no detected');
+
+                await context.report(resource, null, 'The site should be HTTPS');
+
+                return;
+            }
+
+            debug('HTTPS detected');
+
+            this.targetIsServedOverHTTPS = true;
+
+            return;
+        };
+
+        const reportInsecureHops = (response: Response): Array<Promise<void>> => {
+            const { hops } = response;
+
+            return hops.map((hop) => {
+                const fails: boolean = !reportedUrls.has(hop) && !isHTTPS(hop);
+
+                if (fails) {
+                    reportedUrls.add(hop);
+
+                    return context.report(hop, null, `Shouldn't be redirected from HTTP`);
+                }
+
+                return Promise.resolve();
+            });
+        };
+
+        const validateFetchEnd = async (fetchEnd: FetchEnd) => {
+            // We are assuming the first `fetch::end` event recieved is the target url.
+            if (!target) {
+                target = fetchEnd.resource;
+
+                await validateTarget(fetchEnd);
+
+                return;
+            }
+
+            if (!this.targetIsServedOverHTTPS) {
+                return;
+            }
+
+            const { resource, response } = fetchEnd;
+
+            await Promise.all(reportInsecureHops(response));
+
+            if (!reportedUrls.has(resource) && !isHTTPS(resource) && !isDataURI(resource)) {
+                reportedUrls.add(resource);
+
+                await context.report(resource, null, 'Should be served over HTTPS');
+            }
+        };
+
+        /** Returns an array with all the URLs in the given `srcset` attribute or an empty string if none. */
+        const parseSrcSet = (srcset: string): Array<string> => {
+            if (!srcset) {
+                return [];
+            }
+
+            const urls = srcset
+                .split(',')
+                .map((entry) => {
+                    return entry.split(' ')[0].trim();
```

Comment:
you should `trim` entry, before the `split(' ')` to be sure you don't have any whitespace at the beginning of the string.